### ISSUE TITLE
Engine.run with multiple dataloader

### ISSUE THREAD

**Issue by Soo95** [pytorch/ignite#629]
Hi, I'm implementing semi-supervised learning.
In training, I want to train with 'label data' and 'non-label data' sequentially per each iteration.

However, 'label data' and 'non-label data' are unbalanced.
let say if label:non_label = 9:1
In pytorch, I may solve this

```
lable_loader_iter = iter(lable_loader)
non_label_loader_iter = iter(non_label_loader)

for idx in len(label_loader):
   try:
        image, mask = lable_loader_iter.next()
   except:
        somthing;
   try:
        image= non_label_loader_iter .next()
   except:
        somthing;
```

But I couldn't figure out how I can do this in ignite.
In Engine.run, 'data' argument accepts 'list' or 'dataloader'.

Is it possible to use two iterable data in ignite?

**Comment by Soo95** [comment]
Ahh I may use Events.ITERATION_COMPLETED!
I'll try

**Comment by vfdev-5** [comment]
@Soo95 maybe the 1st snippet of this answer can help : https://github.com/pytorch/ignite/issues/574#issuecomment-517992456

**Comment by Soo95** [comment]
@vfdev-5 
Thanks a lot! I'll try
